src/qibo/ui/mpldrawer.py

```diff
@@ -257,13 +261,21 @@ def _draw_controls(ax, i, gate, labels, gate_grid, wire_grid, plot_params, measu
                 "RBS",
                 "ECR",
                 "MS",
+                "UNITARY",
             ]:
 
                 symbol = SYMBOLS.get(name, name)
 
                 if is_dagger:
                     symbol += r"$\rm{^{\dagger}}$"
 
+                if name == "UNITARY" and target.count("_") > 1:
+                    u_gate_single_hash = controls[cci].split("_")[2]
+                    subindex = plot_params["hash_unitary_gates"][u_gate_single_hash]
+                    symbol = r"$\rm_{}$".format(subindex) + symbol
```

Comment:
Here, and in every other place handling subscripts, I'd suggest you to keep a literal pair of braces (that you need to escape as double braces `{{}}`, for a total of 3 pairs...).
```suggestion
                    symbol = r"$\rm_{{{}}}$".format(subindex) + symbol
```

The reason is that your format may be otherwise screwed for multiple digits (or characters, in general)
- `$U_{42}$`: $U_{42}$
- `$U_42$`: $U_42$